### PR TITLE
[fastio] Add fastio port file

### PR DESCRIPTION
**Pull request by YexuanXiao**
- #### What does your PR fix?
 Add new port fastio: https://github.com/cppfastio/fast_io

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
 Yes

### CODE REVIEW

File: ports/fastio/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
+vcpkg_from_github(
```
Comment (LilyWangLL) [inline ports/fastio/portfile.cmake:1]:
```suggestion
# header-only library
vcpkg_from_github(
```

File: ports/fastio/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+
+file(INSTALL ${SOURCE_PATH}/license.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (LilyWangLL) [inline ports/fastio/portfile.cmake:17]:
```suggestion
file(INSTALL "${SOURCE_PATH}/license.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/fastio/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "fastio",
+  "version": "2022-11-14",
```
Comment (LilyWangLL) [inline ports/fastio/vcpkg.json:3]:
```suggestion
  "version-date": "2022-11-14",
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
> After committing all other changes, the version database must be updated
> ```
> git add -u && git commit
> git checkout 6f7ffeb18f99796233b958aaaf14ec7bd4fb64b2 -- versions
> ./vcpkg x-add-version --all
> ```
Thanks for your PR, please also do this to update version database.

**Comment by LilyWangLL** [comment]
> > After committing all other changes, the version database must be updated
> > ```
> > git add -u && git commit
> > git checkout 6f7ffeb18f99796233b958aaaf14ec7bd4fb64b2 -- versions
> > ./vcpkg x-add-version --all
> > ```
> 
> Thanks for your PR, please also do this to update version database.

Ping @YexuanXiao for response, could you please apply this suggestion?

**Review by LilyWangLL — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**